watertap/property_models/coagulation_prop_pack.py

```diff
@@ -0,0 +1,640 @@
+            sf = iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'H2O'], default=1, warning=True)
+            iscale.set_scaling_factor(self.flow_mass_phase_comp['Liq', 'H2O'], sf)
+
+        if iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'TSS']) is None:
+            sf = iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'TSS'], default=1e2, warning=True)
+            iscale.set_scaling_factor(self.flow_mass_phase_comp['Liq', 'TSS'], sf)
+
+        if iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'TDS']) is None:
+            sf = iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'TDS'], default=1e2, warning=True)
+            iscale.set_scaling_factor(self.flow_mass_phase_comp['Liq', 'TDS'], sf)
+
+        if iscale.get_scaling_factor(self.flow_mass_phase_comp['Sol', 'Sludge']) is None:
+            sf = iscale.get_scaling_factor(self.flow_mass_phase_comp['Sol', 'Sludge'], default=1e3, warning=True)
+            iscale.set_scaling_factor(self.flow_mass_phase_comp['Sol', 'Sludge'], sf)
+
+        # these variables do not typically require user input,
+        # will not override if the user does provide the scaling factor
+
+        if self.is_property_constructed('mass_frac_phase_comp'):
+            for pair in self.seawater_mass_frac_dict.keys():
+                if iscale.get_scaling_factor(self.mass_frac_phase_comp[pair]) is None:
+                    if pair[1] == 'H2O':
+                        iscale.set_scaling_factor(self.mass_frac_phase_comp[pair], 1)
+                        # transforming constraints
+                        iscale.constraint_scaling_transform(self.eq_mass_frac_phase_comp[pair], 1)
+                    elif pair[1] == 'Sludge':
+                        iscale.set_scaling_factor(self.mass_frac_phase_comp[pair], 1)
+                        # transforming constraints
+                        iscale.constraint_scaling_transform(self.eq_mass_frac_phase_comp[pair], 1)
+                    else:
+                        sf = (iscale.get_scaling_factor(self.flow_mass_phase_comp[pair])
+                              / iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'H2O']))
+                        iscale.set_scaling_factor(self.mass_frac_phase_comp[pair], sf)
+                        # transforming constraints
+                        iscale.constraint_scaling_transform(self.eq_mass_frac_phase_comp[pair], sf/10)
+
+        if self.is_property_constructed('dens_mass_phase'):
+            if iscale.get_scaling_factor(self.dens_mass_phase) is None:
+                iscale.set_scaling_factor(self.dens_mass_phase, 1e-3)
+
+            # transforming constraints
+            sf = iscale.get_scaling_factor(self.dens_mass_phase)
+            iscale.constraint_scaling_transform(self.eq_dens_mass_phase['Liq'], sf)
+            iscale.constraint_scaling_transform(self.eq_dens_mass_phase['Sol'], sf)
```

Comment:
The constraint transformation should not be in the conditional checking if the scaling factor exists. Otherwise if the scaling factor does exist, then the constraint won't be transformed. Usually, I have a section setting all the scaling factors and then a section for transforming the constraints.